pkg/identityserver/bunstore/invitation_store.go

```diff
@@ -195,58 +194,31 @@ func (s *invitationStore) GetInvitation(ctx context.Context, token string) (*ttn
 		}
 		return nil, err
 	}
-
-	pb, err := invitationToPB(model)
-	if err != nil {
-		return nil, err
-	}
-
-	return pb, nil
+	return invitationToPB(model)
 }
 
 func (s *invitationStore) SetInvitationAcceptedBy(
-	ctx context.Context, token string, accepter *ttnpb.UserIdentifiers,
+	ctx context.Context, token string, usrIDs *ttnpb.UserIdentifiers,
 ) error {
 	ctx, span := tracer.StartFromContext(ctx, "SetInvitationAcceptedBy", trace.WithAttributes(
 		attribute.String("invitation_token", token),
-		attribute.String("user_id", accepter.GetUserId()),
+		attribute.String("user_id", usrIDs.GetUserId()),
 	))
 	defer span.End()
 
-	model, err := s.getInvitationModelBy(ctx, func(q *bun.SelectQuery) *bun.SelectQuery {
-		return q.Where("token = ?", token)
-	})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			return store.ErrInvitationNotFound.WithAttributes(
-				"invitation_token", token,
-			)
-		}
-		return err
-	}
-
-	if model.ExpiresAt != nil && model.ExpiresAt.Before(s.now()) {
-		return store.ErrInvitationExpired.WithAttributes("invitation_token", token)
-	}
-
-	if model.AcceptedByID != nil {
-		return store.ErrInvitationAlreadyUsed.WithAttributes("invitation_token", token)
-	}
-
-	_, userUUID, err := s.getEntity(ctx, accepter)
+	_, userUUID, err := s.getEntity(ctx, usrIDs)
```

Comment:
The advantage of `getInvitationModelBy` is that it maintains tenant barriers based on the context. Our current update statement does not, and I am not sure if tokens are guaranteed to be globally unique.

I think we should keep the old logic which retrieved the entity, even if we take some of the business logic out of it.

Reply:
Checked just in case and you are right, token aren't globally unique, they are separated by tenant barriers. 
Changed to use the model returned from the `getInvitationModelBy`

Reply:
You are skipping the error handling for `getInvitationModelBy`.